compiler/codegen/OMRCodeGenerator_inlines.hpp

```diff
@@ -24,10 +24,19 @@
 
 #include "codegen/OMRCodeGenerator.hpp"
 
-TR::CodeGenerator*
-OMR::CodeGenerator::self()
+inline TR::CodeGenerator* OMR::CodeGenerator::self()
    {
    return static_cast<TR::CodeGenerator*>(this);
    }
 
+inline bool OMR::CodeGenerator::intrinsicFunctionSupported(TR::RecognizedMethod m)
+   {
+   return TR::CodeGenerator::ilOpCodeIsSupported(TR::CodeGenerator::ilOpCodeForIntrinsicFunction(m));
+   }
+
+inline bool OMR::CodeGenerator::suppressInliningOfRecognizedMethod(TR::RecognizedMethod method)
```

Comment:
Why should the code generator care about this? Shouldn't the inliner contain such a query and possibly delegate a question to the code generator if it needs help? i.e. shouldn't this function be defined in the inliner as follows:

```
inline bool OMR::CodeGenerator::suppressInliningOfRecognizedMethod(TR::RecognizedMethod method)
   {
   return cg->intrinsicFunctionSupported(method);
   }
```

Reply:
Or I suppose now that `intrinsicFunctionSupported` is static it would be defined exactly the same, except in the inliner which cares about inlining rather than the code generator which cares about how to evaluate IL to produce machine code.

Reply:
Inliner, or Optimizer in general, need consult CodeGen whether certain feature is supported/enabled unless the feature is assumed to be supported by all CodeGen. In this case, reasons behind whether to suppress inlining of a recognized method vary so that it make sense for CodeGen to have a dedicated query.